force-app/main/adapter/in/sobjects/contact/ContactService.cls

```diff
@@ -208,71 +208,88 @@ public with sharing class ContactService {
             return;
         }
 
-        List<Id> contactIds = new List<Id>();
         Map<Id, Account> accountByPrimaryContactId = new Map<Id, Account>();
         List<Contact> contacts = new List<Contact>();
 
-        for (Contact contactRecord : contactsWithoutAccounts) {
-            //track which contacts are being inserted vs. updated
-            if (contactRecord.Id != null) {
-                contactIds.add(contactRecord.Id);
-            }
-        }
+        //track which contacts are being inserted vs. updated
```

Comment:
In a `BeforeInsert` context, this list `contactIds` will be empty, and if it's empty we shouldn't be trying to query for the households that the contacts(household members) belong in. It seems that this flow only benefits Legacy HouseHolds since it's called in its `BeforeUpdate` context.